test/with_dummyserver/test_proxy_poolmanager.py

```diff
@@ -651,6 +655,93 @@ def test_basic_ipv6_proxy(self) -> None:
 
 
 class TestHTTPSProxyVerification:
+    @staticmethod
+    def _get_proxy_fingerprint_md5(ca_path: str) -> str:
+        proxy_pem_path = pathlib.Path(ca_path).parent / "proxy.pem"
+        proxy_der = ssl.PEM_cert_to_DER_cert(proxy_pem_path.read_text())
+        proxy_hashed = hashlib.md5(proxy_der).digest()
+        fingerprint = binascii.hexlify(proxy_hashed).decode("ascii")
+        return fingerprint
+
+    @staticmethod
+    def _get_certificate_formatted_proxy_host(host: str) -> str:
+        try:
+            addr = ipaddress.ip_address(host)
+        except ValueError:
+            return host
+
+        if addr.version != 6:
+            return host
+
+        # Transform ipv6 like '::1' to 0:0:0:0:0:0:0:1 via '0000:0000:0000:0000:0000:0000:0000:0001'
+        return addr.exploded.replace("0000", "0").replace("000", "")
```

Comment:
Doesn't ``addr.compressed`` achieve this? What am I missing?

Reply:
The loopback address that shows up in the certificate is `0:0:0:0:0:0:0:1` but Python only gives me the options via method to get `::1` or `0000:0000:0000:0000:0000:0000:0000:0001'` so this logic is in here to get the matching certificate one.

Not sure if `0:0:0:0:0:0:0:1` format has a name.

Reply:
This is a decent feature request for Python I think.

Reply:
Alternatively I could just write the assert as `hostname \\'{proxy_hostname}\\' doesn\\'t match` and ignore the certificate side.

It doesn't really matter any which way.